src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -147,27 +149,37 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
         // Internal for unit testing
         internal MemoryCacheEntryOptions GetMemoryCacheEntryOptions()
         {
+            var hasEvictionCriteria = false;
             var options = new MemoryCacheEntryOptions();
             if (ExpiresOn != null)
             {
+                hasEvictionCriteria = true;
                 options.SetAbsoluteExpiration(ExpiresOn.Value);
             }
 
             if (ExpiresAfter != null)
             {
+                hasEvictionCriteria = true;
                 options.SetAbsoluteExpiration(ExpiresAfter.Value);
             }
 
             if (ExpiresSliding != null)
             {
+                hasEvictionCriteria = true;
                 options.SetSlidingExpiration(ExpiresSliding.Value);
             }
 
             if (Priority != null)
             {
+                hasEvictionCriteria = true;
                 options.SetPriority(Priority.Value);
             }
 
+            if (!hasEvictionCriteria)
```

Comment:
I don't think you can have a generic eviction policy for the memory cache that is effective. That would be a separate and orthogonal thing, but IMO the policy should be set by the component that makes use of the cache, as it is that component the one that knows about the parameters of the things that its saving on it.